core/src/main/java/org/mule/runtime/core/util/collection/EventToMessageSequenceSplittingStrategy.java

```diff
@@ -39,35 +46,56 @@ public EventToMessageSequenceSplittingStrategy(ExpressionSplittingStrategy expre
   @Override
   @SuppressWarnings({"unchecked", "rawtypes"})
   public MessageSequence<?> split(Event event) {
-    Message msg = event.getMessage();
-    Object payload = msg.getPayload().getValue();
-    if (payload instanceof MessageSequence<?>) {
-      return ((MessageSequence<?>) payload);
-    }
-    if (payload instanceof Iterator<?>) {
-      return new IteratorMessageSequence(((Iterator<Object>) payload));
+    if (expressionSplitterStrategy.hasDefaultExpression()) {
+      Message msg = event.getMessage();
+      Object payload = msg.getPayload().getValue();
+      if (payload instanceof MessageSequence<?>) {
+        return ((MessageSequence<?>) payload);
+      }
+      if (payload instanceof Iterator<?>) {
+        return new IteratorMessageSequence(((Iterator<Object>) payload));
+      }
+      if (payload instanceof Collection) {
+        return new CollectionMessageSequence(copyCollection((Collection) payload));
+      } else if (payload instanceof CursorIteratorProvider) {
+        return new IteratorMessageSequence(((CursorIteratorProvider) payload).openCursor());
+      } else if (payload instanceof Iterable<?>) {
+        return new IteratorMessageSequence(((Iterable<Object>) payload).iterator());
+      } else if (payload instanceof Object[]) {
+        return new ArrayMessageSequence((Object[]) payload);
+      } else if (payload instanceof NodeList) {
+        return new NodeListMessageSequence((NodeList) payload);
+      } else if (payload instanceof Map<?, ?>) {
+        List<Object> list = new LinkedList<>();
+        Set<Map.Entry<?, ?>> set = ((Map) payload).entrySet();
+        for (Map.Entry<?, ?> entry : set) {
+          // TODO MULE-9502 Support "key" flowVar with MapSplitter in Mule 4
```

Comment:
Can't we remove this TODO and close this JIRA issue if the new approach is to split into entries and not values?